lib/beaker-hostgenerator/data.rb

```diff
@@ -1,5 +1,19 @@
 module BeakerHostGenerator
+  # Contains all the information that ends up in the generated hosts
+  # configuration. This includes the various OS-specific platform
+  # configuration, PE-specific installation & upgrade configuration, and
+  # Beaker-supported roles.
+  #
+  # Any data used by any hypervisor or any other abstraction should be defined
+  # in this module, likely in the `osinfo` hash. The hypervisor implementation
+  # must then use the provided module functions (likely `get_platform_info`) to
+  # extract the relevant portions of the `osinfo` data.
+  #
+  # This module is intended to be used by either directly accessing the static
+  # functions like `BeakerHostGenerator::Data.<function>()` or as a mixin via
+  # `include BeakerHostGenerator::Data` and then `<function>()`.
   module Data
+    module_function
```

Comment:
This `module_function` makes all the methods defined in this module directly accessible as static functions like `BeakerHostGenerator::Data.pe_version` and also as mixin methods accessible via `include BeakerHostGenerator::Data` and then `pe_version`.